sql: windows build broken by go:embed and filepath.Join

The windows build fails on master with the following message:

```
 % cockroach.exe
panic: error loading stopwords: open stopwords\danish.stop: file does not exist

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/util/tsearch.init.0()
        github.com/cockroachdb/cockroach/pkg/util/tsearch/stopwords.go:36 +0x334
```

See https://github.com/golang/go/issues/45230 for the reason.

Jira issue: CRDB-26783

--- a/pkg/util/tsearch/stopwords.go
+++ b/pkg/util/tsearch/stopwords.go
@@ -13,7 +13,7 @@ package tsearch
 import (
 	"bytes"
 	"embed"
-	"path/filepath"
+	"path"
 	"strings"
 )
 
@@ -31,7 +31,9 @@ func init() {
 	for _, f := range dir {
 		filename := f.Name()
 		name := strings.TrimSuffix(filename, ".stop")
-		contents, err := stopwordFS.ReadFile(filepath.Join("stopwords", filename))
+		// N.B. we use path.Join here instead of filepath.Join because go:embed
+		// always uses forward slashes. https://github.com/golang/go/issues/45230
+		contents, err := stopwordFS.ReadFile(path.Join("stopwords", filename))
 		if err != nil {
 			panic("error loading stopwords: " + err.Error())
 		}

